rpmlint/checks/FilelistCheck.py

```diff
@@ -0,0 +1,410 @@
+
+    @staticmethod
+    def isdebuginfo(pkg):
+        return (pkg.name.endswith('-debuginfo') or
+                pkg.name.endswith('-debuginfo-32bit') or
+                pkg.name.endswith('-debuginfo-64bit') or
+                pkg.name.endswith('-debugsource') or pkg.name.endswith('-debug'))
+
+    @staticmethod
+    def notsymlink(pkg, f):
+        mode = pkg.files()[f].mode
+        type = (mode >> 12) & 0o17
+        return type != 0o12
+
+    @staticmethod
+    def ghostfile(pkg, f):
+        ghosts = pkg.ghostFiles()
+        return f in ghosts
+
+    def __init__(self, config, output):
+        super().__init__(config, output)
+
+        # computed from goodprefixes.
+        # Directories that are only allowed to have defined subdirs (such as /usr)
+        self._restricteddirs = set()
+        self._restricteddirs.add('/')
+
+        self.init_goodprefixes()
+        self.init_checks()
+
+        for d in self._goodprefixes:
+            if d.count('/') > 2:
+                self._restricteddirs.add(d[0:-1].rpartition('/')[0])
+
+        for check in self._checks:
+            if 'good' in check:
+                for i in range(len(check['good'])):
+                    pattern = check['good'][i]
+                    if '*' in pattern:
+                        r = fnmatch.translate(pattern)
+                        check['good'][i] = re.compile(r)
+
+            if 'bad' in check:
+                for i in range(len(check['bad'])):
+                    pattern = check['bad'][i]
+                    if '*' in pattern:
+                        r = fnmatch.translate(pattern)
+                        check['bad'][i] = re.compile(r)
+
+    def init_goodprefixes(self):
+        self._goodprefixes = (
+            '/bin/',
+            '/boot/',
+            '/etc/',
+            '/lib/',
+            '/lib64/',
+            '/media/',
+            # SUSE policy handled in separate check
+            '/opt/',
+            '/sbin/',
+            '/srv/',
+            # SUSE policy handled in separate check
+            '/usr/X11R6/',
+            '/usr/bin/',
+            '/usr/etc/',
+            '/usr/games/',
+            '/usr/include/',
+            '/usr/lib/',
+            '/usr/lib64/',
+            '/usr/sbin/',
+            '/usr/share/',
+            # actually only linux is allowed by fhs
+            '/usr/src/linux',
+            '/usr/src/debug/',
+            '/usr/src/packages/',
+            '/var/account/',
+            '/var/cache/',
+            '/var/crash/',
+            '/var/games/',
+            '/var/lib/',
+            '/var/lock/',
+            '/var/log/',
+            '/var/mail/',
+            '/var/opt/',
+            '/var/run/',
+            '/var/spool/',
+            '/var/yp/',
+            # those are not in FHS!
+            '/var/adm/',
+            '/var/nis/',
+            '/emul/',
+            '/run/',
+        )
+
+    def init_checks(self):
+        self._checks = [
```

Comment:
This whole crazy dictionary could be for readability converted in toml file(s) specifying the format as it really does not need to be in here.

Reply:
Good idea. Will we then use 'ignorefileif' with string names of functions that will then found and called?

Reply:
It's doable easily with `getattr(self, 'notsymlink')`.